modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/fragment/renderer/SpacesSectionFragmentRenderer.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.site.cms.site.initializer.internal.fragment.renderer;
+
+import com.liferay.fragment.renderer.FragmentRenderer;
+import com.liferay.fragment.renderer.FragmentRendererContext;
+import com.liferay.frontend.taglib.react.servlet.taglib.ComponentTag;
+import com.liferay.portal.kernel.feature.flag.FeatureFlagManagerUtil;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.GroupConstants;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.taglib.servlet.PageContextFactoryUtil;
+
+import java.io.IOException;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Jürgen Kappler
+ */
+@Component(service = FragmentRenderer.class)
+public class SpacesSectionFragmentRenderer implements FragmentRenderer {
+
+	@Override
+	public String getCollectionKey() {
+		return "sections";
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "Spaces");
+	}
+
+	@Override
+	public boolean isSelectable(HttpServletRequest httpServletRequest) {
```

Comment:
Hi @ambrinchaudhary @jkappler ,
This method is a duplicate code for all other FragmentRenderer in com.liferay.site.cms.site.initializer.internal.fragment.renderer. Can this be extracted into an abstract parent class?

Reply:
Totally @balazssk, it would be much cleaner

Reply:
Done @balazssk

Reply:
thank you very much @jkappler @balazssk !!